src/gas-price/gas-price.ts

```diff
@@ -0,0 +1,236 @@
+    delete gasPriceStore[chainId]![providerName]!.lastOnChainDataFeedValues[dataFeedId];
+  }
+};
+
+// TODO: Copied from airnode-utilities, should we import instead?
+export const getPercentile = (percentile: number, array: ethers.BigNumber[]) => {
+  if (array.length === 0) return;
+
+  array.sort((a, b) => (a.gt(b) ? 1 : -1));
+  const index = Math.ceil(array.length * (percentile / 100)) - 1;
+  return array[index];
+};
+
+export const multiplyGasPrice = (gasPrice: ethers.BigNumber, gasPriceMultiplier: number) =>
+  gasPrice.mul(ethers.BigNumber.from(Math.round(gasPriceMultiplier * 100))).div(ethers.BigNumber.from(100));
+
+/**
+ * Calculates the multiplier to use for pending transactions.
+ * @param recommendedGasPriceMultiplier
+ * @param scalingMultiplier
+ * @param lag
+ * @param scalingWindow
+ * @returns
+ */
+export const calculateScalingMultiplier = (
+  recommendedGasPriceMultiplier: number,
+  scalingMultiplier: number,
+  lag: number,
+  scalingWindow: number
+) => recommendedGasPriceMultiplier + (scalingMultiplier - recommendedGasPriceMultiplier) * (lag / scalingWindow);
+
+/**
+ * Fetches the provider recommended gas price and saves it in the store.
+ * @param chainId
+ * @param providerName
+ * @param rpcUrl
+ * @returns {ethers.BigNumber}
+ */
+export const updateGasPriceStore = async (chainId: string, providerName: string, rpcUrl: string) => {
+  const provider = new ethers.providers.StaticJsonRpcProvider(rpcUrl, {
+    chainId: Number.parseInt(chainId, 10),
+    name: chainId,
+  });
+
+  // Get the provider recommended gas price
+  const goGasPrice = await go(async () => provider.getGasPrice(), { retries: 1, attemptTimeoutMs: 2000 });
+  if (!goGasPrice.success) {
+    // eslint-disable-next-line no-console
+    console.log(`Failed to get provider gas price. Error: ${goGasPrice.error.message}.`);
+    throw goGasPrice.error;
+  }
+
+  // Save the new provider recommended gas price to the state
+  setStoreGasPrices(chainId, providerName, goGasPrice.data);
+
+  return goGasPrice.data;
+};
+
+/**
+ * Fetches the provider recommended gas price, saves it in the store and clears out expired gas prices.
+ * @param chainId
+ * @param providerName
+ * @param rpcUrl
+ */
+export const gasPriceCollector = async (
+  chainId: string,
+  providerName: string,
+  rpcUrl: string,
+  sanitizationSamplingWindow: number
+) => {
+  // Initialize the gas store for the chain if not already present
+  initializeGasStore(chainId, providerName);
+  clearExpiredStoreGasPrices(chainId, providerName, sanitizationSamplingWindow);
+  await updateGasPriceStore(chainId, providerName, rpcUrl);
+};
+
+/**
+ *  Calculates the gas price to be used in a transaction based on sanitization and scaling settings.
+ * @param chainId
+ * @param providerName
+ * @param rpcUrl
+ * @param gasSettings
+ * @param nonce
+ * @returns {ethers.BigNumber}
+ */
+export const airseekerV2ProviderRecommendedGasPrice = async (
```

Comment:
I'd just call this `getAirseekerRecommendedGasPrice`, but I understand why you called it that way.